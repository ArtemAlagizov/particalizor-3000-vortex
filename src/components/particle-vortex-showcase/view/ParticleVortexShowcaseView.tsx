import React from "react";
import { IParticleVortexShowcaseView } from "./IParticleVortexShowcaseView";
import { ParticleVortex } from "particalizor-3000";
import { Styled } from "../styled";

const ParticleVortexShowcaseView: React.FC<IParticleVortexShowcaseView> = ({
  particleNumberValue,
  particleTraceWidthValue,
  particleLifeTimeValue,
  imageWidth,
  imageHeight,
  backgroundColor,
  vortexNumberValue
}) => {
  return (
    <Styled.ShowcaseComponentContainer>
      <ParticleVortex
        imageWidth={imageWidth}
        imageHeight={imageHeight}
        vortexNumber={vortexNumberValue}
        particleTraceWidth={particleTraceWidthValue}
        particleNumber={particleNumberValue}
        particleLifeTime={particleLifeTimeValue}
        backgroundColor={backgroundColor}
      />
    </Styled.ShowcaseComponentContainer>
  );
};

export default ParticleVortexShowcaseView;
