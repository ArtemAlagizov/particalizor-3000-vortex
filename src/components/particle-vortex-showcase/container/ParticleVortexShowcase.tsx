import React, { useState } from "react";
import ParticleVortexShowcaseView from "../view/ParticleVortexShowcaseView";

const ParticleVortexShowcase: React.FC = ({}) => {
  const [
    selectedParticleNumberValue,
    setSelectedParticleNumberValue
  ] = useState<number>(30);
  const [
    selectedParticleTraceWidthValue,
    setSelectedParticleTraceWidthValue
  ] = useState<number>(600);
  const [
    selectedParticleLifeTimeValue,
    setSelectedParticleLifeTimeValue
  ] = useState<number>(1100);
  const [selectedVortexNumberValue, setSelectedVortexNumberValue] = useState<
    number
  >(3);
  const [selectedImageHeight, setSelectedImageHeight] = useState<number>(500);
  const [selectedImageWidth, setSelectedImageWidth] = useState<number>(900);
  const [selectedBackgroundColor, setSelectedBackgroundColor] = useState<
    string
  >("#33344c");

  const onChange = (value: number) =>
    setSelectedParticleNumberValue(value as number);

  const onTraceWidthChange = (value: number) =>
    setSelectedParticleTraceWidthValue(value as number);

  const onLifeTimeChange = (value: number) =>
    setSelectedParticleLifeTimeValue(value as number);

  const onVortexNumberChange = (value: number) =>
    setSelectedVortexNumberValue(value as number);

  const onImageWidthChange = (value: number) =>
    setSelectedImageWidth(value as number);

  const onImageHeightChange = (value: number) =>
    setSelectedImageHeight(value as number);

  const onBackgroundColorChange = (value: string) =>
    setSelectedBackgroundColor(value as string);

  return (
    <ParticleVortexShowcaseView
      particleLifeTimeValue={selectedParticleLifeTimeValue}
      particleNumberValue={selectedParticleNumberValue}
      particleTraceWidthValue={selectedParticleTraceWidthValue}
      vortexNumberValue={selectedVortexNumberValue}
      backgroundColor={selectedBackgroundColor}
      imageWidth={selectedImageWidth}
      imageHeight={selectedImageHeight}
    />
  );
};

export default ParticleVortexShowcase;
