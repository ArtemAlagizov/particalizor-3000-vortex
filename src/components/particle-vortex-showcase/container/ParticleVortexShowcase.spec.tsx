import React from "react";
import { mount, shallow } from "enzyme";
import ParticleVortexShowcase from "./ParticleVortexShowcase";
import ParticleVortexShowcaseView from "../view/ParticleVortexShowcaseView";

describe("Given a ParticleVortexShowcase", () => {
  describe("when rendering", () => {
    it("should match snapshot", () => {
      const component = shallow(
        <ParticleVortexShowcase
          backgroundColor={""}
          particleTraceWidth={1}
          particleNumber={1}
          particleLifeTime={1}
          vortexNumber={1}
        />
      );

      expect(component).toMatchSnapshot();
    });
    it("should render ParticleVortexShowcaseView", () => {
      const component = mount(
        <ParticleVortexShowcase
          vortexNumber={1}
          particleLifeTime={1}
          particleNumber={1}
          particleTraceWidth={1}
          backgroundColor={""}
        />
      );

      expect(component.find(ParticleVortexShowcaseView).length).toBe(1);
    });
  });
});
