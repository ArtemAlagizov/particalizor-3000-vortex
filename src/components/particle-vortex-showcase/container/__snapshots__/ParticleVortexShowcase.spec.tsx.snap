// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ParticleVortexShowcase when rendering should match snapshot 1`] = `ShallowWrapper {}`;
